### ISSUE TITLE
running as standalone application?

### ISSUE THREAD

**Issue by stelehm** [matsim-org/pt2matsim#24]
I am trying to follow the instructions given in the readme.
After downloading the binary-file "pt2matsim-17.12.jar", tried to create a default config file with running the following command from the command line:
`
java -cp pt2matsim-17.12.jar org.matsim.pt2matsim.run.CreateDefaultOsmConfig`

This only results in the following error-message:

> 
> Error: A JNI error has occurred, please check your installation and try again
> Exception in thread "main" java.lang.NoClassDefFoundError: org/matsim/core/config/ConfigGroup
>         at java.lang.Class.getDeclaredMethods0(Native Method)
>         at java.lang.Class.privateGetDeclaredMethods(Unknown Source)
>         at java.lang.Class.privateGetMethodRecursive(Unknown Source)
>         at java.lang.Class.getMethod0(Unknown Source)
>         at java.lang.Class.getMethod(Unknown Source)
>         at sun.launcher.LauncherHelper.validateMainClass(Unknown Source)
>         at sun.launcher.LauncherHelper.checkAndLoadMain(Unknown Source)
> Caused by: java.lang.ClassNotFoundException: org.matsim.core.config.ConfigGroup
>         at java.net.URLClassLoader.findClass(Unknown Source)
>         at java.lang.ClassLoader.loadClass(Unknown Source)
>         at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
>         at java.lang.ClassLoader.loadClass(Unknown Source)
>         ... 7 more

Did I miss a step, or did I do something fundamentally wrong?

**Comment by polettif** [comment]
The jar you downloaded is not really a standalone file as it does not include all the dependencies. For example MATSim itself is not included, which explains the you came across here. The readme is a bit misleading since it's actually easiest to include pt2matsim as a maven dependency (by adding the snippet from the readme to pom.xml file). I'll look into an actual standalone build.

**Comment by polettif** [comment]
[Release 18.1](https://bintray.com/polettif/matsim/pt2matsim/18.1#files/org%2Fmatsim%2Fpt2matsim%2F18.1) contains pt2matsim-18.1-shaded.jar which can be used as a standalone.

**Comment by stelehm** [comment]
Hello and thank you for the fast respond.
I downloaded the standalone jar and tried to run 
`java -cp pt2matsim-18.1-shaded.jar org.matsim.pt2matsim.run.CreateDefaultOsmConfig`

It produces following error:

```
INFO Config:131 context=[file:/U:/Transportation%20modeling/4th%20step/preprocessing/network/]
WARN Config:312 Module "linkStats" is removed manually from config
WARN Config:312 Module "subtourModeChoice" is removed manually from config
WARN Config:312 Module "TimeAllocationMutator" is removed manually from config
WARN Config:312 Module "counts" is removed manually from config
WARN Config:312 Module "vehicles" is removed manually from config
WARN Config:312 Module "global" is removed manually from config
WARN Config:312 Module "network" is removed manually from config
WARN Config:312 Module "planCalcScore" is removed manually from config
WARN Config:312 Module "scenario" is removed manually from config
WARN Config:312 Module "travelTimeCalculator" is removed manually from config
WARN Config:312 Module "ptCounts" is removed manually from config
WARN Config:312 Module "JDEQSim" is removed manually from config
WARN Config:312 Module "changeMode" is removed manually from config
WARN Config:312 Module "vspExperimental" is removed manually from config
WARN Config:312 Module "planscalcroute" is removed manually from config
WARN Config:312 Module "transitRouter" is removed manually from config
WARN Config:312 Module "parallelEventHandling" is removed manually from config
WARN Config:312 Module "households" is removed manually from config
WARN Config:312 Module "changeLegMode" is removed manually from config
WARN Config:312 Module "qsim" is removed manually from config
WARN Config:312 Module "plans" is removed manually from config
WARN Config:312 Module "transit" is removed manually from config
WARN Config:312 Module "controler" is removed manually from config
WARN Config:312 Module "facilities" is removed manually from config
WARN Config:312 Module "strategy" is removed manually from config
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 0
        at org.matsim.pt2matsim.run.CreateDefaultOsmConfig.main(CreateDefaultOsmConfig.java:48)
```

**Comment by polettif** [comment]
You have to provide a path to a file which should be crated as a program argument. E.g:
`java -cp pt2matsim-18.1-shaded.jar org.matsim.pt2matsim.run.CreateDefaultOsmConfig defaultConfig.xml`
(The warnings can be ignored)

**Comment by stelehm** [comment]
Thank you again. That was obviously my fault, sorry for that question.
I managed to create the config.xml, but now I get an error when creating the network:

> 2018-01-30 11:20:47,261  WARN MGC:175 Assuming that coordinates are in longitude first notation, i.e. (longitude, latitude).
Exception in thread "main" java.lang.IllegalArgumentException: org.opengis.referencing.NoSuchAuthorityCodeException: No code "EPSG:4326" from authority "EPSG" found for object of type "EngineeringCRS".
        at org.matsim.core.utils.geometry.geotools.MGC.getCRS(MGC.java:178)
        at org.matsim.core.utils.geometry.transformations.GeotoolsTransformation.<init>(GeotoolsTransformation.java:57)
        at org.matsim.core.utils.geometry.transformations.TransformationFactory.getCoordinateTransformation(TransformationFactory.java:80)
        at org.matsim.pt2matsim.osm.OsmMultimodalNetworkConverter.convert(OsmMultimodalNetworkConverter.java:85)
        at org.matsim.pt2matsim.run.Osm2MultimodalNetwork.run(Osm2MultimodalNetwork.java:89)
        at org.matsim.pt2matsim.run.Osm2MultimodalNetwork.run(Osm2MultimodalNetwork.java:66)
        at org.matsim.pt2matsim.run.Osm2MultimodalNetwork.main(Osm2MultimodalNetwork.java:49)
Caused by: org.opengis.referencing.NoSuchAuthorityCodeException: No code "EPSG:4326" from authority "EPSG" found for object of type "EngineeringCRS".
        at org.geotools.referencing.factory.epsg.CartesianAuthorityFactory.noSuchAuthorityException(CartesianAuthorityFactory.java:136)
        at org.geotools.referencing.factory.epsg.CartesianAuthorityFactory.createEngineeringCRS(CartesianAuthorityFactory.java:130)
        at org.geotools.referencing.factory.epsg.CartesianAuthorityFactory.createCoordinateReferenceSystem(CartesianAuthorityFactory.java:121)
        at org.geotools.referencing.factory.AuthorityFactoryAdapter.createCoordinateReferenceSystem(AuthorityFactoryAdapter.java:801)
        at org.geotools.referencing.factory.ThreadedAuthorityFactory.createCoordinateReferenceSystem(ThreadedAuthorityFactory.java:731)
        at org.geotools.referencing.DefaultAuthorityFactory.createCoordinateReferenceSystem(DefaultAuthorityFactory.java:179)
        at org.geotools.referencing.CRS.decode(CRS.java:520)
        at org.matsim.core.utils.geometry.geotools.MGC.getCRS(MGC.java:176)
        ... 6 more

There seems to be an issue with the coordinate transformation. (Which is weird, because I let `<param name="outputCoordinateSystem" value="null" />`  and there shouldn't be any transformation)
I am not sure if this is a problem with the standalone-jar, the code or if I am using it wrong.

**Comment by polettif** [comment]
This seems to be a problem with the standalone jar.

**Comment by polettif** [comment]
Should be fixed with v18.1.1
